tests/unit/test_repo_manager.py

```diff
@@ -34,179 +62,253 @@ def readRepo(self):
         repo.open("r")
         return repo
 
+    def init(self):
+        self.runCommand("init {}".format(self._repoPath))
+
+    def addDataset(self):
+        self._datasetName = "test_dataset"
+        cmd = "add-dataset {} {}".format(self._repoPath, self._datasetName)
+        self.runCommand(cmd)
+
+    def addReferenceSet(self):
+        self._referenceSetName = "test_rs"
+        fastaFile = paths.ncbi37FaPath
+        self.runCommand("add-referenceset {} {} --name={}".format(
+            self._repoPath, fastaFile, self._referenceSetName))
+
+    def addReadGroupSet(self):
+        bamFile = paths.bamPath
+        self._readGroupSetName = "test_rgs"
+        cmd = (
+            "add-readgroupset {} {} {} --referenceSetName={} "
+            "--name={}").format(
+            self._repoPath, self._datasetName, bamFile,
+            self._referenceSetName, self._readGroupSetName)
+        self.runCommand(cmd)
+
 
 class TestAddDataset(AbstractRepoManagerTest):
 
     def setUp(self):
         super(TestAddDataset, self).setUp()
-        self.runCommand("init {}".format(self._repoPath))
+        self.init()
 
-    def testAddDataset(self):
+    def testDefaults(self):
         name = "test_dataset"
         self.runCommand("add-dataset {} {}".format(self._repoPath, name))
         repo = self.readRepo()
         dataset = repo.getDatasetByName(name)
         self.assertEqual(dataset.getLocalId(), name)
 
-    def testAddDatasetWithSameName(self):
+    def testSameName(self):
         name = "test_dataset"
         cmd = "add-dataset {} {}".format(self._repoPath, name)
         self.runCommand(cmd)
         self.assertRaises(
+            exceptions.DuplicateNameException, self.runCommand, cmd)
+
+
+class TestAddReferenceSet(AbstractRepoManagerTest):
+
+    def setUp(self):
+        super(TestAddReferenceSet, self).setUp()
+        self.init()
+
+    def testDefaults(self):
+        fastaFile = paths.ncbi37FaPath
+        name = os.path.split(fastaFile)[1].split(".")[0]
+        self.runCommand("add-referenceset {} {}".format(
+            self._repoPath, fastaFile))
+        repo = self.readRepo()
+        referenceSet = repo.getReferenceSetByName(name)
```

Comment:
I disagree here I'm afraid. I don't see what we would have to gain by allowing for duplicate names within a container. The schema _allows_ for duplicate names (bizarrely, IMO), but it doesn't _require_ it. Our lives are made a lot easier by regarding names as within-container, user specified IDs, and we're free to do that. Maybe I'm misunderstanding the point you're making here though? Feel free to open an issue to discuss it in more detail.